server/games/game.py

```diff
@@ -271,13 +272,15 @@ def add_game_connection(self, game_connection):
 
                 if self.desyncs > 20:
                     await self.mark_invalid(ValidityState.TOO_MANY_DESYNCS)
+                    return
```

Comment:
Due to this bug, you probably want to write a DB migration script to delete records from `game_player_stats` that refer to invalid games.